src/Raven.Client/Documents/Operations/Indexes/SetIndexesLockOperation.cs

```diff
@@ -34,6 +37,18 @@ public SetIndexesLockOperation(Parameters parameters)
                 throw new ArgumentNullException(nameof(parameters.IndexNames));
 
             _parameters = parameters;
+
+            FilterAutoIndexes();
+        }
+
+        private void FilterAutoIndexes()
+        {
+            _parameters.IndexNames = _parameters.IndexNames.Where(indexName => indexName.StartsWith("Auto/", StringComparison.OrdinalIgnoreCase) == false).ToArray();
+            
+            if (_parameters.IndexNames.Length == 0)
```

Comment:
Better to throw when there any any auto indexes in the list instead of silently filtering them out:
```
var anyAutoIndexes = _parameters.IndexNames
.Where(indexName => indexName.StartsWith("Auto/", StringComparison.OrdinalIgnoreCase)
.Any();
```

Reply:
If I throw - as you request - then when the list that is passed contains a mixture of both static & auto indexes then work will Not be done at all for the static indexes

Is that the wanted flow?
The way it is now - work is still done for the static-indexes

What do you think ?

Reply:
Answered on Slack